src/site/_transforms/responsive-images/helpers.js

```diff
@@ -0,0 +1,42 @@
+const site = require("../../_data/site");
+const path = require("path");
+
+/**
+ *
+ * @param {string} src src attribute for image
+ * @param {string} outputPath Output path for HTML file
+ *
+ * @return {{src: string, isLocal: boolean}}
+ */
+function determineImagePath(src, outputPath) {
+  const isLocal = !RegExp("^https?://").test(src);
+
+  // If an image has a protocol then we should just return the src as is.
+  // If the outputPath is false then it means the file has permalink set to
+  // false. In this scenario we should also just return the src as is.
+  if (!isLocal || !outputPath) {
+    return {
+      src,
+      isLocal,
+    };
+  }
+
+  if (path.isAbsolute(src)) {
+    return {src: new URL(src, site.imageCdn), isLocal};
+  }
+
+  // At this point we've determined that the image has a relative path.
+  let base = path
+    .dirname(outputPath)
+    .split(path.sep)
+    .pop();
+
+  // The handbook is the only place on the site where we nest paths.
```

Comment:
But it's pretty easy to make more of these, even accidentally. Can we just set the path based on the input? I don't know what "outputPath" here is exactly.